clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -858,6 +859,27 @@ class PackDeductionScope {
       Info.PendingDeducedPacks[Pack.Index] = Pack.Outer;
   }
 
+  // Return the size of the saved packs if all of them has the same size.
+  std::optional<unsigned> getSavedPackSizeIfAllEqual() const {
+    if (Packs.size() == 0 ||
+        Packs[0].Saved.getKind() != clang::TemplateArgument::Pack)
```

Comment:
when is `Packs[0].Saved.getKind() != clang::TemplateArgument::Pack` true?

Reply:
for that matter when is `Packs.size() != 0` ?

Reply:
The reason I added them is because I wanted to be careful. What if someone constructs a `PackDeductionScope` with a non pack `TemplateArgument`.

Should I replace them with insertions?

Reply:
I think you can remove it entirely, there is an assert in `.pack_size();` - that should simplify that code a bit !

Reply:
Removed this precondition. I have also removed a redundant condition from the checking of the equivalence of pack sizes.